apps/meteor/app/api/server/v1/chat.js

```diff
@@ -217,6 +217,14 @@ API.v1.addRoute(
 				throw new Meteor.Error('error-invalid-params', 'The "message" parameter must be provided.');
 			}
 
+			const { rid: roomId } = this.bodyParams.message;
+
+			const subscription = Subscriptions.findOneByRoomIdAndUserId(roomId, this.userId);
+
+			if (!subscription) {
```

Comment:
why not `canSendMessage` function instead?

Reply:
`canSendMessage` or `canAccessRoom` would return `true` to all public rooms (even when the user is not a member of the room). Since in this case we'd like to block users from sending messages to rooms they are not part of, I guess none of these would fit 🤔

Reply:
correct me if i'm wrong, but being able to send a message on a channel i don't belong ,yet, (but have permission) should cause an auto join. this is more like a feature than a bug in my opinion

maybe you have more information on what caused or caused this issue/confusion?

just an image to describe where this feature could be used:
<img width="536" alt="image" src="https://user-images.githubusercontent.com/5263975/185037670-33da483d-1f93-438f-a84b-0063217b12ad.png">